code/modules/mob/dead/observer/say.dm

```diff
@@ -27,10 +27,22 @@
 	return "[pick("whines", "cries", "spooks", "complains", "drones", "mutters")], \"[text]\"";
 
 /mob/dead/observer/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq)
-	if(radio_freq)
-		speaker = speaker.GetSource()
+	if (isnull(client))
+		return
+
+	if (isnull(radio_freq))
+		if (!(client.prefs.toggles & CHAT_GHOSTEARS))

```

Comment:
But that's not what Ghostears does, I'm pretty sure.
It's a toggle between global and local chat, not chat and nochat.